src/pudl/extract/ferc714.py

```diff
@@ -1,72 +1,97 @@
+            "encoding": "iso-8859-1",
+        },
+        "interchange_ba_ferc714": {
+            "name": "Part 2 Schedule 5 - Balancing Authority Interchange.csv",
+            "encoding": "iso-8859-1",
+        },
+        "lambda_hourly_ba_ferc714": {
+            "name": "Part 2 Schedule 6 - Balancing Authority Hourly System Lambda.csv",
+            "encoding": "utf-8",
+        },
+        "lambda_description_ferc714": {
+            "name": "Part 2 Schedule 6 - System Lambda Description.csv",
+            "encoding": "iso-8859-1",
+        },
+        "description_pa_ferc714": {
+            "name": "Part 3 Schedule 1 - Planning Area Description.csv",
+            "encoding": "iso-8859-1",
+        },
+        "demand_forecast_pa_ferc714": {
+            "name": "Part 3 Schedule 2 - Planning Area Forecast Demand.csv",
+            "encoding": "utf-8",
+        },
+        "demand_hourly_pa_ferc714": {
+            "name": "Part 3 Schedule 2 - Planning Area Hourly Demand.csv",
+            "encoding": "utf-8",
+        },
+        "respondent_id_ferc714": {
+            "name": "Respondent IDs.csv",
+            "encoding": "utf-8",
+        },
+    }
+)
+"""Dictionary mapping PUDL tables to FERC-714 filenames and character encodings."""
 
-TABLE_ENCODING = {
-    "id_certification_ferc714": "iso-8859-1",
-    "gen_plants_ba_ferc714": "iso-8859-1",
-    "demand_monthly_ba_ferc714": None,
-    "net_energy_load_ba_ferc714": None,
-    "adjacency_ba_ferc714": "iso-8859-1",
-    "interchange_ba_ferc714": "iso-8859-1",
-    "lambda_hourly_ba_ferc714": None,
-    "lambda_description_ferc714": "iso-8859-1",
-    "description_pa_ferc714": "iso-8859-1",
-    "demand_forecast_pa_ferc714": None,
-    "demand_hourly_pa_ferc714": None,
-    "respondent_id_ferc714": None,
-}
-"""Dictionary describing the character encodings of the FERC 714 CSV files."""
 
-
-def extract(
-    ferc714_settings: Ferc714Settings = Ferc714Settings(), pudl_settings=None, ds=None
-):
+@multi_asset(
+    outs={"raw_" + table_name: AssetOut() for table_name in FERC714_FILES},
+    required_resource_keys={"datastore"},
+)
+def extract_ferc714(context):
     """Extract the raw FERC Form 714 dataframes from their original CSV files.
 
     Args:
-        ferc714_settings: Object containing validated settings relevant to
-            FERC Form 714.
-        pudl_settings (dict): A PUDL settings dictionary.
-        ds (Datastore): instance of the datastore
+        context: dagster keyword that provides access to resources and config.
 
     Returns:
-        dict: A dictionary of dataframes, with raw FERC 714 table names as the
-        keys, and minimally processed pandas.DataFrame instances as the values.
+        A tuple of extracted FERC-714 dataframes.
     """
     warnings.warn(
         "Integration of FERC 714 into PUDL is still experimental and incomplete.\n"
         "The data has not yet been validated, and the structure may change."
     )
```

Comment:
Make this a new and different logging message about how folks will get all of the years of data.